doc/storage/api/api.rst

```diff
@@ -649,17 +680,144 @@ These definitions must be defined in the header file :file:`psa/protected_storag
+
+   .. retval:: PSA_ERROR_STORAGE_FAILURE
+        The operation failed because the physical storage has failed (Fatal error).     
+
+
+   The function renames ``uid`` to ``uid_new`` retaining the storage flags  that ``uid`` was created with. 
+
+   If the caller specifies ``PSA_STORAGE_FLAG_OVERWRITE`` the operation atomically replaces the exisitng contents of ``uid_new`` with those of ``uid``.
+
+   However, ``uid`` must not have been created with more permissive Storage flags than ``uid_new``
+        
+   Except in the case of ``PSA_ERROR_STORAGE_FAILURE``, in which case no guarantees can be made, the operation shall either succeed or leave storage unchanged. 
+
+
+
+
+.. function:: psa_ps_iterator_init
+
+   .. summary:: 
+       Initialises an iterator that can be used to return a list of uids in the Protected Storage .
+
+   .. param::  uint8_t* context
+       A pointer to a context for this iterator. This is set to a new value on success and is undefined on error. The content of the iterator is implementation defined.  The pointer MUST be valid and MUST point to a block of memory of the size given by the ``PSA_STORAGE_ITERATOR_CTX__SIZE`` macro to which the caller has write access. 
+   .. param:: psa_storage_uid_t mask
+       A value used to filter the results included in this iteration. 
+   .. param:: psa_storage_uid_t filter
+       A value used to filter the results included in this iteration. 
+
+   .. return:: psa_status_t
+       A status indicating the success or failure of the operation.
+
+   .. retval:: PSA_SUCCESS
+       The operation completed successfully.
+   .. retval:: PSA_ERROR_STORAGE_FAILURE
+       The operation failed because the physical storage has failed (Fatal error).
+   .. retval:: PSA_ERROR_INVALID_ARGUMENT
+       The operation failed because the caller cannot access some or all of the memory in the range [``context``, ``context + PSA_STORAGE_ITERATOR_CTX__SIZE - 1``].    
+
+
+   Any bit positions where the mask is set to one must match the value in the filter.
+   
+   That is, the iterator returns those values where ``uid & mask = filter``
+
+   The iterator will only returns those ``uid`` that were created by the caller. It MUST not return any ``uid`` created by a different user. 
+
+   An iterator is not required to return uids in any specific order, but MUST return them in a consistent order each time it is called. For example, if an implementation returns entries in numerical order, it should not arbitrarily change to returning them in creation order. However, the caller should not make assumptions as to the order in which entries are returned, except that each uid will be returned only once in each iteration.
+
+   Changes to storage by other users MUST NOT affect any open iterations. 
+
+   A caller may initialize multiple iteration contexts at the same time. Each iteration shall be independent. Calling ``psa_ps_iterator_next()`` on one iterator MUST not effect any other open iteration. 
```

Comment:
This is a critical part of the specification - combined with the absence of a 'resource release' function, it indicates [implicitly] that there cannot be any RoT service-side state associated with an iterator, without causing an unmanageable resource exhaustion problem.